js/DropZone/DropZoneComponent.js

```diff
@@ -0,0 +1,564 @@
+            this.classManager.update(this.body, [
+                instance.options.interactionClasses.windowEnter,
+                instance.options.interactionClasses.dropZoneEnter
+            ]);
+        } else {
+            this.throwValidationError(text, id);
+            this.classManager.update(this.body, [
+                instance.options.interactionClasses.windowEnter,
+                instance.options.interactionClasses.dropZoneEnter,
+                instance.options.interactionClasses.dropZoneError
+            ]);
+        }
+
+        // call any additional callbacks passed in via options
+        if (instance.options.customDropZoneEnter && typeof instance.options.customDropZoneEnter === 'function') {
+            instance.options.customDropZoneEnter.apply(this, [].slice.call(arguments));
+        }
+    }
+
+    /**
+     * Handle DropZone dropZoneLeave callback
+     * @param {Boolean} valid
+     * @param {String} text
+     * @param {DropZone} instance
+     */
+    handleDropzoneLeave ({ valid, text, instance }) {
+        const id = instance.getDropZoneId();
+
+        // update helper text
+        if (valid) {
+            this.updateInfoState(id, instance.options.windowEnterHtml);
+            this.classManager.update(this.body, [instance.options.interactionClasses.windowEnter]);
+        } else {
+            this.throwValidationError(text, id);
+            this.classManager.update(this.body, [
+                instance.options.interactionClasses.windowEnter,
+                instance.options.interactionClasses.dropZoneError
+            ]);
+        }
+
+        // call any additional callbacks passed in via options
+        if (instance.options.customDropZoneLeave && typeof instance.options.customDropZoneLeave === 'function') {
+            instance.options.customDropZoneLeave.apply(this, [].slice.call(arguments));
+        }
+    }
+
+    /**
+     * Handle DropZone dropZoneEnter callback
+     * @param {Array} files
+     * @param {Boolean} valid
+     * @param {String} text
+     * @param {DropZone} instance
+     */
+    handleDropZoneDrop ({ files, valid, text, instance }) {
+        const id = instance.getDropZoneId();
+
+        if (valid) {
+            this.updateDropZoneFiles(id);
+            this.classManager.update(this.body, [instance.options.interactionClasses.dropZoneSuccess]);
+        } else {
+            // If a persist property has been set on the file, we will throw a
+            // validation error which will persist once the file is dropped, this
+            // helps us out when we need to throw an error when a user uses an associated
+            // native file input's "browse files"
+            let persist = false;
+            // a crude shim for Array.find, we just need to know if any of the files
+            // have the persist flag set to true
+            files.forEach(file => !persist && file.meta.persist ? persist = true : null);
```

Comment:
I've simplified this by ditching the find shim altogether in favour of `_.find(...)`